File: other/sorts/quick.js
function qs(array){
    if (array.length <= 1){return array}
    var pivot  = array[0]
    var bottom = array.filter(function(x){return x < pivot})
    var middle = array.filter(function(x){return x == pivot})
    var top    = array.filter(function(x){return x > pivot})
    return qs(bottom).concat(middle.concat(qs(top)))
}

File: other/sorts/quick.rb
def qs array
    return array if array.length <= 1
    pivot  = array.sample

    top    = array.select {|n| n > pivot}
    middle = array.select {|n| n == pivot}
    bottom = array.select {|n| n < pivot}

    qs(bottom) + middle + qs(top)
end
